src/miner.cpp

```diff
@@ -435,6 +424,56 @@ static bool ProcessBlockFound(CBlock* pblock, CWallet& wallet, CReserveKey& rese
     return true;
 }
 
+bool static ScanLoop(CBlock *pblock, CBlockIndex *pindexPrev, CWallet *pwallet, CReserveKey& reservekey)
+{
+    UpdateTime(pblock, pindexPrev);
+    arith_uint256 hashTarget = arith_uint256().SetCompact(pblock->nBits);
+
+    uint256 hash;
+    if (ScanHash(pblock, &hash)) {
+        if (UintToArith256(hash) <= hashTarget) {
+            // Found a solution
+            SetThreadPriority(THREAD_PRIORITY_NORMAL);
+            LogPrintf("BitcoinMiner:\n");
+            LogPrintf("proof-of-work found  \n  hash: %s  \ntarget: %s\n", hash.GetHex(), hashTarget.GetHex());
+            ProcessBlockFound(pblock, *pwallet, reservekey);
+            SetThreadPriority(THREAD_PRIORITY_LOWEST);
+
+            return true;
+        }
+    }
+
+    return false;
+}
+
+bool MineBlock(CWallet *pwallet, uint256& hash)
+{
+    CReserveKey reservekey(pwallet);
+    unsigned int nExtraNonce = 0;
+
+    while (true) {
+        CBlockIndex *pindexPrev;
+
+        auto_ptr<CBlockTemplate> pblocktemplate(CreateNewBlockWithKey(reservekey, pindexPrev));
+        if (!pblocktemplate.get()) {
+            return false;
+        }
+
+        CBlock *pblock = &pblocktemplate->block;
+        IncrementExtraNonce(pblock, pindexPrev, nExtraNonce);
+
+        while (true) {
+            if (ScanLoop(pblock, pindexPrev, pwallet, reservekey)) {
+                hash = pblock->GetHash();
+                return true;
+            }
+            boost::this_thread::interruption_point();
+            if (pblock->nNonce >= 0xffff0000)
```

Comment:
I mean, it's just a nit and probably the more irrational one of the 3, I would just like to minimize the accesses to nBits and nNonce in the code so that the strictly-pow stuff doesn't take much to read.
If you don't make anything about this, thank you for the try.